dask/dataframe/core.py

```diff
@@ -3316,16 +3316,26 @@ def cummin(self, axis=None, skipna=True, out=None):
         )
 
     def _validate_condition(self, cond):
-        if not (
-            is_dask_collection(cond)
-            or is_dataframe_like(cond)
-            or is_series_like(cond)
-            or is_index_like(cond)
-        ):
+        def is_alignable_collection(x):
+            return (
+                is_dask_collection(x)
+                or is_dataframe_like(x)
+                or is_series_like(x)
+                or is_index_like(x)
+            )
+
+        if not (is_alignable_collection(cond) or callable(cond)):
             raise ValueError(
                 f"Condition should be an object that can be aligned with {self.__class__}, "
-                f" which includes Dask or pandas collections, DataFrames or Series."
+                f" which includes Dask or pandas collections, DataFrames or Series, or a Callable."
             )
+        if callable(cond):
+            result = cond(self._meta)
+            if not is_alignable_collection(result):
+                raise ValueError(
+                    f"Condition should be an object that can be aligned with {self.__class__}, "
+                    f" which includes Dask or pandas collections, DataFrames or Series, or a Callable."
+                )
```

Comment:
You could save a few loc and avoid repeating the error with something like:
```py
cond_result = cond(self._meta) if callable(cond) else cond
if not is_alignable_collection(cond_result):
    ...
```

Reply:
Good point, will do.